nntrainer/dataset/random_data_producers.cpp

```diff
@@ -0,0 +1,167 @@
+public:
+  /**
+   * @brief Construct a new props data size object with a default value
+   *
+   * @param value default value
+   */
+  PropsDataSize(unsigned int value = 512) :
+    nntrainer::Property<unsigned int>(value) {}
+  static constexpr const char *key = "size"; /**< unique key to access */
+  using prop_tag = uint_prop_tag;            /**< property type */
+};
+
+RandomDataOneHotProducer::RandomDataOneHotProducer() :
+  rd_one_hot_props(new Props()) {}
+
+RandomDataOneHotProducer::~RandomDataOneHotProducer() {}
+
+const std::string RandomDataOneHotProducer::getType() const {
+  return RandomDataOneHotProducer::type;
+}
+
+unsigned long long
+RandomDataOneHotProducer::size(const std::vector<TensorDim> &input,
+                               const std::vector<TensorDim> &output) const {
+  return std::get<PropsDataSize>(*rd_one_hot_props).get();
+}
+
+void RandomDataOneHotProducer::setProperty(
+  const std::vector<std::string> &properties) {
+  auto left = loadProperties(properties, *rd_one_hot_props);
+  NNTR_THROW_IF(!left.empty(), std::invalid_argument)
+    << "There are unparsed properties, size: " << left.size();
+}
+
+DataProducer::Gernerator
+RandomDataOneHotProducer::finalize(const std::vector<TensorDim> &input_dims,
+                                   const std::vector<TensorDim> &label_dims) {
+  /** check if the given producer is ready to finalize */
+  auto &[min_, max_, sz] = *rd_one_hot_props;
+
+  NNTR_THROW_IF(std::any_of(label_dims.begin(), label_dims.end(),
+                            [](const TensorDim &dim) {
+                              return dim.channel() != 1 || dim.height() != 1;
+                            }),
+                std::invalid_argument)
+    << "Label dimension containing channel or height not allowed";
+
+  NNTR_THROW_IF(min_.get() > max_.get(), std::invalid_argument)
+    << "Min value is bigger then max value, min: " << min_.get()
+    << "max: " << max_.get();
+
+  NNTR_THROW_IF(size(input_dims, label_dims) == 0, std::invalid_argument)
+    << "size is zero, dataproducer does not provide anything";
+
+  /** prepare states for the generator */
+  std::vector<std::uniform_int_distribution<unsigned int>> label_chooser_;
+  label_chooser_.reserve(label_dims.size());
+  std::transform(label_dims.begin(), label_dims.end(),
+                 std::back_inserter(label_chooser_),
+                 [this](const TensorDim &label_dim) {
+                   return std::uniform_int_distribution<unsigned int>(
+                     0, label_dim.width() - 1);
+                 });
+
+  std::mt19937 rng;
+  rng.seed(getSeed());
+
+  /** DataProducer::Generator */
+  return [rng, sz = size(input_dims, input_dims), input_dims, label_dims,
+          min_ = min_.get(), max_ = max_.get(), current_iteration = 0ULL,
+          label_chooser = std::move(label_chooser_)]() mutable {
+    if (current_iteration++ == sz) {
+      current_iteration = 0;
+      return DataProducer::Iteration(true, {}, {});
+    }
+
```

Comment:
I wonder if this random data is going to converge. I mean.. there is no relation between data and label.

Reply:
Yes, this is just a sample whoever can use if they built a model but not data, but want to run to see if their model has a nyproblem.

Reply:
> Yes, this is just a sample whoever can use if they built a model but not data, but want to run to see if their model has a nyproblem.

so.. I guess.. it is not guarantee to converge while training with this dataset. So.. it is not for training.. it is just for the test of generation of data?? On second thought.. then why do we need this?.. just functional test? But with this kind of dataset, we do not test except semantics.. I guess. Is it the reason to support this?

Reply:
The intention here is that, as databuffer is going through a big change, this kind of simple implementation works as a reference. If I am to test on a batch queue or databuffer, we'd better test on a simplest possible implementation on the designed interface, so that the given data producer would have no problem but batch queue has problem in case of batch queue test fails.

The refactored databuffers and components like batch queues are tested on this referential implementation without having compilated, possibly problematic producer involved. Because it's simple, it didn't take more than two hours to implement this component.

Reply:
> The intention here is that, as databuffer is going through a big change, this kind of simple implementation works as a reference.

I guess this is matter when we develop the code but not the matter of actual user of nntrainer when the new databuffer is done.  And also if it is just for the test queue or databuffer, it need to be test rather than main feature.

> Because it's simple, it didn't take more than two hours to implement this component.

In my opinion, I fully understand the reason of this ( reducing debugging time and etc. ) but it is not a matter of implementation time issue. I mean, if we are going to support the custom data set ( I guess it is callback feature, correct me if it is not ), this random data set producer is more appropriate for that. Also, we do need example for it too.

Reply:
I see :) I guess we can migrate this to custom example some time soon!
For now, we need a bit more infras to support this as a custom. (like ini parsing and app context registering.)

Reply:
Thanks for your considering. I guess this is ok for now. But once we finished implementation the new dataset, then we need to migrate this to examples.